### PR TITLE
Move `quantum_theta` to `quantum.py`

### LINKED ISSUE #1443: Migrate `quantum_theta` and write tests
The `quantum_theta` function should be moved from `plasmapy.formulary.dimensionless` to `plasmapy.formulary.quantum`.  I would have no problem still keeping it exposed in the `dimensionless` namespace.

Additionally, the one test written for `quantum_theta` is not a real test, so a full set of tests need to be written.

### PR DESCRIPTION
**Pull request by JaydenR2305**
Resolves #1443 

This pull request moves the `quantum_theta` function definition into `quantum.py`, while still exposing the `dimensionless` namespace to the function to ensure no compatibility issues. Additionally, this pull request rewrites the test functions for `quantum_theta` to include a more thorough test suite

- [x] I have added a changelog entry for this pull request.

- [x] If adding new functionality, I have added tests and
      docstrings.

- [x] I have fixed any newly failing tests.

### CODE REVIEW

**Comment by JaydenR2305** [comment]
What should I do about the invalid reference? @StanczakDominik

**Comment by StanczakDominik** [comment]
Oh dear, it's in the changelog for 0.6 and looking for the previous location of this function. Perhaps just change it from
```rst

- Adds formulas (which were missing) to the docstrings of
  `~plasmapy.formulary.dimensionless.quantum_theta` and
  `~plasmapy.formulary.dimensionless.beta`. (`#1041 <https://github.com/plasmapy/plasmapy/pull/1041>`__)
```
to
```rst

- Adds formulas (which were missing) to the docstrings of
  ``~plasmapy.formulary.dimensionless.quantum_theta`` and
  `~plasmapy.formulary.dimensionless.beta`. (`#1041 <https://github.com/plasmapy/plasmapy/pull/1041>`__)
```

because I don't think we want to spend too much time maintaining out-of-date changelogs!

File: changelog/1671.trivial.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Move the `~plasmapy.formulary.dimensionless.quantum_theta` function to the `~plasmapy.formulary.quantum` module. Note, this function can still be called from the `plasmapy.formulary.dimensionless` module without issue.
```
Comment (StanczakDominik) [inline changelog/1671.trivial.rst:1]:
```suggestion
Move the ``~plasmapy.formulary.dimensionless.quantum_theta`` function to the `~plasmapy.formulary.quantum` module. Note, this function can still be called from the `plasmapy.formulary.dimensionless` module without issue.
```
I bet it's going to complain about the missing function here as well!

File: plasmapy/formulary/tests/test_quantum.py [hunk 2]
```diff
@@ -231,3 +232,26 @@ def test_quantum_aliases():
...
+
+    def test_units(self, theta):
+        """Test the return units"""
+
+        assert theta.unit.is_equivalent(u.dimensionless_unscaled)
+
+    def test_value(self, theta):
+        """Compare the calculated theta with the expected value."""
+
+        assert np.isclose(theta.value, 12.72906)
```
Comment (StanczakDominik) [inline plasmapy/formulary/tests/test_quantum.py:257]: Could you think of a couple more test cases to add here? Think numerical stability, weird floating point values, large numbers, small numbers, limiting cases. This function is pretty simple and doesn't need much, but I'd still like to have a couple more test cases for it, so the code is more resilient to possible errors we introduce later. Try using `@pytest.mark.parametrize` with a single test function for that.

**Review by StanczakDominik — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 3]
```diff
@@ -927,7 +927,7 @@ Improved Documentation
 - Put the docstring for `plasmapy.particles.Particle.is_category` into
   `numpydoc` format. (`#1039 <https://github.com/plasmapy/plasmapy/pull/1039>`__)
 - Adds formulas (which were missing) to the docstrings of
-  `~plasmapy.formulary.dimensionless.quantum_theta` and
+  ``~plasmapy.formulary.dimensionless.quantum_theta`` and
```
Comment (namurphy) [inline CHANGELOG.rst:930]:
```suggestion
  ``plasmapy.formulary.dimensionless.quantum_theta`` and
```

File: changelog/1671.trivial.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Move the ``~plasmapy.formulary.dimensionless.quantum_theta`` function to the `~plasmapy.formulary.quantum` module. Note, this function can still be called from the `plasmapy.formulary.dimensionless` module without issue.
```
Comment (namurphy) [inline changelog/1671.trivial.rst:1]:
```suggestion
Move the ``plasmapy.formulary.dimensionless.quantum_theta`` function to the `~plasmapy.formulary.quantum` module. Note, this function can still be called from the `plasmapy.formulary.dimensionless` module without issue.
```

File: docs/whatsnew/0.6.0.rst [hunk 5]
```diff
@@ -184,7 +184,7 @@ Improved Documentation
 - Put the docstring for `plasmapy.particles.particle_class.Particle.is_category` into
   ``numpydoc`` format. (`#1039 <https://github.com/plasmapy/plasmapy/pull/1039>`__)
 - Adds formulas (which were missing) to the docstrings of
-  `~plasmapy.formulary.dimensionless.quantum_theta` and
+  ``~plasmapy.formulary.dimensionless.quantum_theta`` and
```
Comment (namurphy) [inline docs/whatsnew/0.6.0.rst:187]:
```suggestion
  ``plasmapy.formulary.dimensionless.quantum_theta`` and
```

**Review by namurphy — COMMENTED** [review]
A few minor suggestions to remove the tilde that was in the obsolete links.  The tilde would make `~plasmapy.formulary.speeds` show up only as `speeds`, but isn't needed when using double backticks for inline literal text.  

Thank you for doing this!

File: plasmapy/formulary/tests/test_quantum.py (hunk 2 above)
Comment (JaydenR2305) [reply plasmapy/formulary/tests/test_quantum.py]: I added some more test cases, do those look good?

Comment (StanczakDominik) [reply plasmapy/formulary/tests/test_quantum.py]: They look lovely :) Thanks!

**Review by StanczakDominik — APPROVED** [review]
Solid stuff, thanks, @JaydenR2305! :)

**Pull request merged**